source/OutfitterPanel.cpp

```diff
@@ -335,186 +381,362 @@ double OutfitterPanel::DrawDetails(const Point &center)
+		return "No outfit selected.";
+
+	// If outfit is not available in the Outfitter, respond that it can't be bought here.
+	if(!(outfitter.Has(selectedOutfit) || player.Stock(selectedOutfit) > 0))
+	{
+		// The store doesn't have it.
+		return "You cannot buy this outfit here. "
+			"It is being shown in the list because you have one, "
+			"but this " + planet->Noun() + " does not sell them.";
+	}
 
 	// Check special unique outfits, if you already have them.
-	int mapSize = selectedOutfit->Get("map");
-	bool mapMinables = selectedOutfit->Get("map minables");
-	if(mapSize > 0 && player.HasMapped(mapSize, mapMinables))
-		return "You have already mapped all the systems shown by this map, "
-			"so there is no reason to buy another.";
+	// Skip this if told to for sake of speed
+	if(checkSpecialItems)
+	{
+		int mapSize = selectedOutfit->Get("map");
+		bool mapMinables = selectedOutfit->Get("map minables");
+		if(mapSize > 0 && player.HasMapped(mapSize, mapMinables))
+			return "You have already mapped all the systems shown by this map, "
+				"so there is no reason to buy another.";
 
-	if(HasLicense(selectedOutfit->TrueName()))
-		return "You already have one of these licenses, "
-			"so there is no reason to buy another.";
+		if(HasLicense(selectedOutfit->TrueName()))
+			return "You already have one of these licenses, "
+				"so there is no reason to buy another.";
+	}
+
+	// Determine what you will have to pay to buy this outfit.
+	int64_t cost = player.StockDepreciation().Value(selectedOutfit, day);
+	int64_t credits = player.Accounts().Credits();
+	if(cost > credits)
+		return "You don't have enough money to buy this outfit. You need a further " +
+			Format::CreditString(cost - credits);
+
+	// Add the cost to buy the required license.
+	int64_t licenseCost = LicenseCost(selectedOutfit, false);
+	if(cost + licenseCost > credits)
+		return "You don't have enough money to buy this outfit because you also need to buy a "
+			"license for it. You need a further " +
+			Format::CreditString(cost + licenseCost - credits);
 
 	// Check that the player has any necessary licenses.
-	int64_t licenseCost = LicenseCost(selectedOutfit, onlyOwned);
 	if(licenseCost < 0)
 		return "You cannot buy this outfit, because it requires a license that you don't have.";
 
-	// Check if the outfit is available to get at all.
-	bool isInCargo = player.Cargo().Get(selectedOutfit);
-	bool isInStorage = player.Storage().Get(selectedOutfit);
-	bool isInStore = outfitter.Has(selectedOutfit) || player.Stock(selectedOutfit) > 0;
-	if(isInStorage && (onlyOwned || isInStore || playerShip))
+	// The outfit is able to be purchased (available in the Outfitter, licensed and affordable).
+	return true;
+}
+
+
+
+// Checking where it will go (can it fit in fleet cargo), not checking where it will
+// be sourced from.
+ShopPanel::TransactionResult OutfitterPanel::CanFitInCargo(bool returnReason) const
+{
+	// TODO: https://github.com/endless-sky/endless-sky/issues/10599
+
```

Comment:
If this is a todo for you (as in you want to do it in this PR), please don't, this PR probably already is the largest code one we have. If it's a general "we should get to this sooner or later", having an issue open here on Github is enough.
```suggestion
```